scripts/Interface/ui/drill/model/DrillParameter.py

```diff
@@ -0,0 +1,222 @@
+    WORKSPACE_TYPE = "workspace"
+    COMBOBOX_TYPE = "combobox"
+    STRING_TYPE = "string"
+    BOOL_TYPE = "bool"
+    FLOAT_ARRAY_TYPE = "floatArray"
+    INT_ARRAY_TYPE = "intArray"
+
+    """
+    Name of the parameter.
+    """
+    _name = None
+
+    """
+    Value of the parameter.
+    """
+    _value = None
+
+    """
+    Parameter documentation.
+    """
+    _documentation = None
+
+    """
+    Parameter type.
+    """
+    _type = None
+
+    """
+    Parameter allowed values.
+    """
+    _allowedValues = None
+
+    """
+    Reference to the parameter controller.
+    """
+    _controller = None
+
+    """
+    Validation state of the parameter value.
+    """
+    _valid = True
+
+    """
+    Error message associated to an invalid state.
+    """
+    _validationErrorMsg = None
+
+    """
+    Sent when the parameter value has been checked.
+    """
+    checked = Signal()
+
+    """
+    Sent when the parameter value changed.
+    """
+    valueChanged = Signal()
+
+    def __init__(self, name):
+        super().__init__()
+        self._name = name
+        self._validationErrorMsg = ""
+
+    def setController(self, controller):
+        """
+        Set the parameter controller.
+
+        Args:
+            controller (DrillParameterController): controller
+        """
+        self._controller = controller
+
+    def initFromProperty(self, mantidProperty):
+        """
+        Inititialize the parameter from a mantid property. This method populates
+        the type, documentation, allowed values of the parameters. It also sets
+        the value to the property default value.
+
+        Args:
+            mantidProperty (Property): the property
+        """
+        self._documentation = mantidProperty.documentation
+        self._allowedValues = mantidProperty.allowedValues
+        value = mantidProperty.value
+        if (isinstance(mantidProperty, FileProperty)):
+            self._type = self.FILE_TYPE
+        elif (isinstance(mantidProperty, MultipleFileProperty)):
+            self._type = self.FILES_TYPE
+            if not value:
+                value = ""
+        elif (isinstance(mantidProperty, (WorkspaceGroupProperty,
+                                          MatrixWorkspaceProperty))):
+            self._type = self.WORKSPACE_TYPE
+        elif (isinstance(mantidProperty, StringPropertyWithValue)):
+            if mantidProperty.allowedValues:
+                self._type = self.COMBOBOX_TYPE
+            else:
+                self._type = self.STRING_TYPE
+        elif (isinstance(mantidProperty, BoolPropertyWithValue)):
+            self._type = self.BOOL_TYPE
+        elif (isinstance(mantidProperty, FloatArrayProperty)):
+            self._type = self.FLOAT_ARRAY_TYPE
+        elif (isinstance(mantidProperty, IntArrayProperty)):
+            self._type = self.INT_ARRAY_TYPE
+        else:
+            self._type = self.STRING_TYPE
+
+        if isinstance(value, numpy.ndarray):
```

Comment:
I think what is meant here is only 1D array. More than one dimensions should not be allowed.

Reply:
I added a check for the dimension